katsdpimageutils/primary_beam_correction.py

```diff
@@ -115,6 +115,30 @@ def central_freq(path):
     return np.array(c_freq_plane)
 
 
+def check_band_type(path):
+    raw_image = read_fits(path)
+    try:
+        if 'BANDCODE' in raw_image.header:
+            if raw_image.header['BANDCODE'] == 'L':
+                return 'MKAT-AA-L-JIM-2020'
+            if raw_image.header['BANDCODE'] == 'UHF':
+                return 'MKAT-AA-UHF-JIM-2020'
+            if raw_image.header['BANDCODE'] == 'S':
+                return 'MKAT-AA-L-JIM-2020'
+        else:
+            freqs = central_freq(path)
+            start_freq = freqs[0]/1e6
+            end_freq = freqs[-1]/1e6
+            if start_freq >= 856 and end_freq <= 1712:  # L-band
+                return 'MKAT-AA-L-JIM-2020'
+            if start_freq >= 544 and end_freq <= 1087:  # UHF-band
+                return 'MKAT-AA-UHF-JIM-2020'
+            if start_freq >= 2000 and end_freq <= 4000:  # S-band
+                return 'MKAT-AA-L-JIM-2020'
+    except KeyError:
+        logging.error('Exception occurred, keywords not found', exc_info=True)
```

Comment:
I think this function has a few issues:

It will return None after logging the `KeyError` at the end, and this return value will cause another exception further down the line when you call `CircularBeam`. Is this what is expected to happen? 

You should add some comments about S-band defaulting to L-band since we don't have a model for it yet. Perhaps also add a docstring to this function?

They way you are doing it is quite difficult to maintain, since you are repeating the model name 6 times in this function - if this changes you'll have to hack through the code and find all these places the string is hardcoded. My suggestion would be to make a variable at the top of this module, somewhere near line 12 and define a dictionary that maps the band name from header['BANDCODE'] to the model that it will use. Like this:
```
# Mapping from 'BANDCODE' to katbeam model name
# NOTE: S-band defaults to L-band model since katbeam has no S-band models
BAND_MAP = {
    'L': ...,
    'UHF': ...,
    'S': ...
}
```
then in this function you can just lookup the band from your dictionary, defaulting to None if there isn't one:
```
raw_image = read_fits(path)
band = raw_image.header.get('BANDCODE')
model = BAND_MAP.get(band)
```
then you can do the check on the frequency ranges `if model is None`.

I would use some logging in this function as well. Log a warning if the 'BANDCODE' isn't in the FITS header, with mention that frequency ranges are being used to determine the band. Also log at the end before returning, saying which primary beam model is being used and which band was chosen.

Also, I think this function should just return the L-band model as a default (with an appropriate warning log) if it can't find anything (e.g. no/unknown BANDCODE in header, or the frequency ranges are wrong).

Reply:
I understand what you are saying, it makes sense. I tried to address all your comments  on this commit.